### DISCUSSION TITLE
Give users the option to load a subset of encodings to minimize js bundle size

### DISCUSSION THREAD

**Discussion by eekboom** [ashtuchkin/iconv-lite#343] [Ideas]
After adding iconv-lite to my small preact app, the minified js bundly grew by about 300KB.

Is there a way to selectively bundle a few encodings (utf-8, iso-8859-*) and fetch the others when needed?
I guess there isn't because the API to encode/decode is synchronous.

In that case it would be a very welcome feature.

**Comment by ashtuchkin** [comment]
That's correct. I guess a good way to do that would be to split encodings into different npm modules and give users the option to only depend on the one they need. I'll take that into account when planning new version. 

Thank you!

**Comment by bjohansebas** [comment]
This is quite interesting, and yes, I would like that too (reducing builds). Normally, bundlers handle this through tree-shaking. Here, what I wonder is how we could do it. Before submitting a PR, I’d like to first know a good approach that could improve this
